### ISSUE TITLE
ImportError: cannot import name 'URLResolver'

### ISSUE THREAD

**Issue by webconexus** [infoportugal/wagtail-modeltranslation#230]
I am getting this error while trying your module. What is going wrong here?

`Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/core/management/__init__.py", line 338, in execute
    django.setup()
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/apps/registry.py", line 108, in populate
    app_config.import_models()
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/apps/config.py", line 202, in import_models
    self.models_module = import_module(models_module_name)
  File "/home/webconexus/portfolio/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/wagtail/contrib/routable_page/models.py", line 4, in <module>
    from django.urls import URLResolver
ImportError: cannot import name 'URLResolver'`

**Comment by DiogoMarques29** [comment]
Hi @webconexus 
I don't think this problem is related to wagtail-modeltranslation.
Check the compatibility versions of Python, Django and Wagtail:
https://docs.wagtail.io/en/latest/releases/upgrading.html#compatible-django-python-versions
https://docs.djangoproject.com/en/2.1/faq/install/#what-python-version-can-i-use-with-django

Cheers

**Comment by webconexus** [comment]
Okay I updated my Django version. However now I get a different error. Wagtail-modeltranslation does not work with Django 2.1?

`Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/core/management/__init__.py", line 357, in execute
    django.setup()
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/apps/registry.py", line 120, in populate
    app_config.ready()
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/wagtail_modeltranslation/apps.py", line 20, in ready
    from .patch_wagtailadmin import patch_wagtail_models
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/wagtail_modeltranslation/patch_wagtailadmin.py", line 8, in <module>
    from django.core.urlresolvers import reverse
ModuleNotFoundError: No module named 'django.core.urlresolvers'`

**Comment by DiogoMarques29** [comment]
Only version v0.10b1 supports django >= 2.0

**Comment by webconexus** [comment]
Ah ok thanks @DiogoMarques29! What command do I have to run to upgrade wagtail-modeltranslation?

**Comment by DiogoMarques29** [comment]
https://pypi.org/project/wagtail-modeltranslation/0.10b1/

**Comment by webconexus** [comment]
Hmm okay getting weird error again:

`Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/core/management/base.py", line 316, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/core/management/base.py", line 350, in execute
    self.check()
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/core/management/base.py", line 379, in check
    include_deployment_checks=include_deployment_checks,
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/core/management/base.py", line 366, in _run_checks
    return checks.run_checks(**kwargs)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/core/checks/registry.py", line 71, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/wagtail/admin/checks.py", line 63, in get_form_class_check
    if not issubclass(edit_handler.get_form_class(), WagtailAdminPageForm):
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/wagtail/admin/edit_handlers.py", line 345, in get_form_class
    widgets=self.widget_overrides())
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/wagtail/admin/edit_handlers.py", line 63, in get_form_for_model
    return metaclass(class_name, (form_class,), form_class_attrs)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/wagtail/admin/forms/models.py", line 66, in __new__
    new_class = super(WagtailAdminModelFormMetaclass, cls).__new__(cls, name, bases, attrs)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/modelcluster/forms.py", line 241, in __new__
    new_class = super(ClusterFormMetaclass, cls).__new__(cls, name, bases, attrs)
  File "/home/webconexus/portfolio/lib/python3.6/site-packages/django/forms/models.py", line 266, in __new__
    raise FieldError(message)
django.core.exceptions.FieldError: Unknown field(s) (search_description_en, seo_title_en, slug_nl, seo_title_nl, slug_en, search_description_nl) specified for ProjectsPage`


My translation.py : 

`from .models import HomePage
from modeltranslation.translator import TranslationOptions
from modeltranslation.decorators import register

@register(HomePage)
class FooTR(TranslationOptions):
    fields = (
        'intro_left',
    )`

My model: 

`class HomePage(Page):
    # Add the intro fields to homepage model
    intro_heading       = models.CharField(max_length=255, null=True)
    intro_sub_heading   = models.CharField(max_length=255, null=True)
    intro_left          = RichTextField(blank=True)
    intro_right         = RichTextField(blank=True)

    content_panels = Page.content_panels + [
        FieldPanel('intro_heading'),
        FieldPanel('intro_sub_heading'),
        FieldPanel('intro_left', classname="full"),
        FieldPanel('intro_right', classname="full"),
    ]`

Base.py : 

`# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
import os

from django.utils.translation import gettext_lazy as _

LANGUAGES = (
    ('nl', _('Dutch')),
    ('en', _('English')),
)`

Any idea what is going wrong here?

**Comment by DiogoMarques29** [comment]
Have you run `sync_page_translation_fields` and `update_translation_fields`?
Also, do you have `'wagtail_modeltranslation.makemigrations'` and `'wagtail_modeltranslation.migrate'` in the installed apps?

**Comment by webconexus** [comment]
Yes I added both to installed apps. I get this traceback when I try to run `sync_page_translation_fields` and `update_translation_fields`

Both the same error.

**Comment by DiogoMarques29** [comment]
Please show django and wagtail versions

**Comment by webconexus** [comment]
`(portfolio) webconexus@wagtail:~/src/wagtail-portfolio$ pip show django
Name: Django
Version: 2.1.7`

`(portfolio) webconexus@wagtail:~/src/wagtail-portfolio$ pip show wagtail
Name: wagtail
Version: 2.4`

`(portfolio) webconexus@wagtail:~/src/wagtail-portfolio$ python --version
Python 3.6.7`

`(portfolio) webconexus@wagtail:~/src/wagtail-portfolio$ pip show wagtail-modeltranslation
Name: wagtail-modeltranslation
Version: 0.10b1`

**Comment by DiogoMarques29** [comment]
I created a branch on our demo repository with the versions you are using and I was unable to replicate your problem.
Please check: https://github.com/infoportugal/wagtail-modeltranslation-demo/tree/issue_230

**Comment by webconexus** [comment]
Very weird. The traceback is also weird, because I didn't use anything of the app in that model. The traceback is regard ProjectPage. You see down here I don't use any wagtail-modeltranslation there

`from django.db import models
from django import forms

from wagtail.core.models import Page
from wagtail.core.fields import RichTextField
from wagtail.admin.edit_handlers import FieldPanel, MultiFieldPanel, FieldRowPanel
from wagtail.snippets.models import register_snippet
from modelcluster.fields import ParentalKey, ParentalManyToManyField
from wagtail.images.edit_handlers import ImageChooserPanel

from wagtail.snippets.models import register_snippet


class ProjectsPage(Page):
    description = models.CharField(max_length=255, blank=True)

    content_panels = Page.content_panels + [
        FieldPanel('description', classname="full"),
    ]`

**Comment by webconexus** [comment]
Does it have something to do with the get request I use in the model I want to translate?

`class HomePage(Page):
    # Add the intro fields to homepage model
    intro_heading       = models.CharField(max_length=255, null=True)
    intro_sub_heading   = models.CharField(max_length=255, null=True)
    intro_left          = RichTextField(blank=True)
    intro_right         = RichTextField(blank=True)

    content_panels = Page.content_panels + [
        FieldPanel('intro_heading'),
        FieldPanel('intro_sub_heading'),
        FieldPanel('intro_left', classname="full"),
        FieldPanel('intro_right', classname="full"),
    ]

    # Display the portfolio items !IMPORTANT! take a good look at the slug, you have to create a base parent page first (Projects page) and insert te slug under here. Otherwise the page wont render.
    # Generate a projects page and insert that slug down here at Page.objects.get(slug='my-slug')
    def get_context(self, request):
        context                     = super(HomePage, self).get_context(request)
        context['portfolio_page']   = Page.objects.get(slug='afgeronde-projecten')
        return context`

**Comment by DiogoMarques29** [comment]
As soon as you add wagtail-modeltranslation to your project, all pages will have some fields translated, that means that you have to register ALL subclasses of Page defined in your project. You can't have at the moment a subclass of Page that is not registered for translation.

**Comment by webconexus** [comment]
Thankyou for the clarification @DiogoMarques29 all is working fine now 👍

**Comment by webconexus** [comment]
@DiogoMarques29 So I have added some translations, it's going ok. However child pages of my model disappeared in Wagtail Admin. I suggest it's because I did this to avoid the error:

`@register(ProjectsPage)
class ProjectsPageTR(TranslationOptions):
    pass

@register(IndividualProjectsPage)
class IndividualProjectsPageTR(TranslationOptions):
    pass`

How should I solve this? My pages are gone now in admin, but they are still in the database because they are rendering on my webpage.
